### PR TITLE
Gemfile: use current Ruby version

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR changes the Travis execution to use their default Ruby-based build task: "bundle exec rake".

- changed given Ruby version to "the ruby that we run" (this changed the Gemfile.lock, too, which was now locked to 2.3.2, the version I was running; perhaps not what you want)
- rely on Rake default task to run Travis scripts
- add newer Bundler to Gemfile
- add rake to Gemfile

Q: 

- Is there a specific version reason for the one that was in the Gemfile before?

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,12 +1,13 @@
-ruby "2.2.4"
+ruby RUBY_VERSION
```
Comment (timrogers) [inline Gemfile]: I've not seen people do this before - the `ruby` declaration tends to be used as a declaration of the version of Ruby that the project is intended for. I see it as a useful signal of that.

The [Bundler documentation](http://bundler.io/v1.3/gemfile_ruby.html) seems to suggest that this is useful if you depend on features of specific Ruby implementations though (which I'm pressure sure we don't, so maybe this is fine).

Comment (olleolleolle) [inline Gemfile]: @timrogers Hi again, well, it's a not-often-used "hint feature" of Bundler that can be used by gem authors. I documented it here, https://github.com/bundler/bundler/pull/5200 - but with luck the next release of Bundler will do this version navigation unassisted.

Comment (olleolleolle) [inline Gemfile]: (I am sorry to confuse you with that docs link - that is a tricky way to "abuse" this feature.)

Comment (timrogers) [inline Gemfile]: Aha! That helps. Is setting that option really useful here, given that we're only testing against one Ruby and are committing our `Gemfile.lock`? (perhaps a silly question)

Comment (olleolleolle) [inline Gemfile]: It's not a silly question, it's at the heart of the usability of this feature.

To help us always use the exact version, we can change back to exact match, _and_ add a `.ruby-version` file with that ruby version number to the project. That `.ruby-version` is used automatically by the Travis build's `rvm` step.

And it'll also be used by the passing developer, who's trying to assist this project.

I'll change the file contents!

**Comment by olleolleolle** [comment]
@timrogers I'll rebase and make it refer to our latest understanding.

**Comment by olleolleolle** [comment]
It'll be clearer in another PR. :-|

**Comment by timrogers** [comment]
Thanks for your work on this :heart:

**Pull request closed without merging**